File: app/src/main/java/ru/starksoft/differ/sample/base/viewmodel/mutable/BaseMutableParam.java
package ru.starksoft.differ.sample.base.viewmodel.mutable;

import org.jetbrains.annotations.NotNull;

abstract class BaseMutableParam {

	@Override
	public int hashCode() {
		throw new UnsupportedOperationException("You can't use get() for mutable param!");
	}

	@NotNull
	@Override
	public String toString() {
		return "BaseMutableParam{}";
	}

}

File: diff-adapter/src/main/java/ru/starksoft/differ/mutable/BaseMutableParam.kt
package ru.starksoft.differ.mutable

@Suppress("EqualsOrHashCode")
abstract class BaseMutableParam {

	override fun hashCode(): Int {
		throw UnsupportedOperationException("You can't use get() for mutable param!")
	}

	override fun toString(): String {
		return "BaseMutableParam{}"
	}
}
